packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -0,0 +1,151 @@
+import {ArrayValue, NumberValue, RecordValue, Schema } from "@coveo/bueno";
+import {validateOptions} from '../../../../utils/validate-payload';
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {buildController, Controller} from '../../../controller/headless-controller';
+import {loadReducerError} from '../../../../utils/errors';
+import {cartReducer as cart} from '../../../../features/commerce/context/cart/cart-slice';
+import {
+  addCartItem,
+  removeCartItem,
+  setCart,
+  updateCartItemQuantity
+} from '../../../../features/commerce/context/cart/cart-actions';
+
+const optionsSchema = new Schema({
+  cart: new ArrayValue({
+    each: new RecordValue({
+      values: {
+        product: new RecordValue({
+          options: {required: true}
+        }),
+        quantity: new NumberValue({
+          required: true,
+          min: 1,
+        })
+      }
+    }),
+  }),
+});
+
+export interface CartOptions {
+  cart?: CartItem[];
+}
+
+export interface CartItem {
+  product: Product;
+  quantity: number;
+}
+
+export interface Product {
```

Comment:
Are these properties really all optional ? Even `sku` ?

Reply:
Yes, according to the API spec, they're all optional: https://docs.google.com/spreadsheets/d/1-WFpnipA3fgwCX3GhvnmJvPSwO_jfiQY-lZytsapfKs/edit?pli=1#gid=0&range=D20:D22